### PR TITLE
Add spell checking

### PR DESCRIPTION
**Pull request by sedders123**
This PR follows the approach of https://github.com/srobo/website/pull/384 to add `cspell` to the docs.

### CODE REVIEW

File: .spelling [hunk 1]
```diff
@@ -0,0 +1,49 @@
+APRILTAG
+BODMAS
+camcon
```
Comment (PeterJCLaw) [inline .spelling:3]: This is a shortening of "Cambridge Connectors" and may even be a contraction which is unique to SR. Maaaybe stylise as "CamCon"?

File: .spelling [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+DataTraveler
+Duratool
+DTSE9H
+Ebuyer
+imax
```
Comment (PeterJCLaw) [inline .spelling:8]:
```suggestion
iMAX
```

File: .spelling [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Nikitin
+ODROID
+overcurrent
```
Comment (PeterJCLaw) [inline .spelling:11]: Hrm, is this really a word? I'd be tempted to hyphenate this.

File: .spelling [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+Ruggeduinos
+Scarzy
+srobo
+ST7202USBGB
+studentrobotics
+usbkey
+Webots
+Zoloto
+
+# Markup
```
Comment (PeterJCLaw) [inline .spelling:26]: :+1: to breaking out sections.

File: Rakefile [hunk 5]
```diff
@@ -84,4 +90,8 @@ task :validate_sidebar_tree => [:build] do
   puts "Sidebar links validated successfully"
 end
 
-task :validate => [:validate_kit_versions, :validate_links, :validate_sidebar_tree]
+task :validate_spellings => [:build_spellings] do
+  sh('npm run spell-check')
+end
+
+task :validate => [:validate_kit_versions, :validate_links, :validate_sidebar_tree, :validate_spellings]
```
Comment (PeterJCLaw) [inline Rakefile:97]: Thought for another time: I wonder if CI should check each of these independently? (Currently I suspect a failure in one will mask a failure in another).

File: package.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "scripts": {
+    "test": "npm run spell-check",
+    "spell-check": "cspell \"**/*.html\" \"**/*.md\" \"**/*.js\" \"**/*.css\""
```
Comment (PeterJCLaw) [inline package.json:4]: Thoughts on pushing the file patterns into the cspell config? That ensures that running it directly on the command line or via an editor integration picks up only the "right" files. Unless it's intentional that IDE integrations check everything?
(I suspect I missed this on the website repo)

File: programming/sr/ruggeduinos/index.md [hunk 7]
```diff
@@ -41,7 +41,7 @@ R.ruggeduinos["752303138333517171B1"].something...
 </div>
 
 
-[Setting pin modes](#pinmodes) {#pinmodes}
+[Setting pin modes](#pin-modes) {#pin-modes}
```
Comment (PeterJCLaw) [inline programming/sr/ruggeduinos/index.md:44]: Probably worth checking for anything outside this repo which might link here. Ideally urls shouldn't change.

**Review by PeterJCLaw — COMMENTED** [review]
Nice, thanks for putting this together. Some small comments inline.

File: .spelling (hunk 3 above)
Comment (sedders123) [reply .spelling:11]: It's got a [wikipedia](https://en.wikipedia.org/wiki/Overcurrent) and seems to be in a standard ([IEC 60364](https://webstore.iec.ch/preview/info_iec60364-1%7Bed5.0%7Den_d.pdf) 131.4) so think its ok to be a word.

File: .spelling [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+Duratool
+DTSE9H
+Ebuyer
+imax
+Nikitin
+ODROID
+overcurrent
+piezo
+Plazma
+PULLUP
```
[the remark this answers is not in the document]
Comment (sedders123) [reply .spelling:14]: This was from sample code

File: resources/troubleshooter/data.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
-    "askiffixed_question": {
+    "ask_if_fixed_question": {
```
[the remark this answers is not in the document]
Comment (sedders123) [reply resources/troubleshooter/data.json:2]: This was find and replace :)

File: programming/sr/ruggeduinos/index.md (hunk 7 above)
Comment (sedders123) [reply programming/sr/ruggeduinos/index.md:44]: No exact matches on links to #pinmodes are indexed by Google, Bing, DuckDuckGo, or Yandex. Agree changing URLs aren't great but this feels like an improvement. Happy to switch back if others disagree though :)

File: package.json (hunk 6 above)
Comment (sedders123) [reply package.json]: 👍 [9ad5664](https://github.com/srobo/docs/pull/320/commits/9ad566462e812aaefbdeac1a2ac511ce36f6d8e0)

File: .spelling (hunk 1 above)
Comment (sedders123) [reply .spelling]: 👍 [28792c8](https://github.com/srobo/docs/pull/320/commits/28792c8b10a1ab0c78a4bb044a6f63bfa085eddf)

File: programming/sr/ruggeduinos/index.md (hunk 7 above)
Comment (PeterJCLaw) [reply programming/sr/ruggeduinos/index.md:44]: :shipit:

**Review by trickeydan — APPROVED** [review]
🎉 Nice one

**Pull request merged**